### DISCUSSION TITLE
ImGui::LogFinish() newline is redundant

### DISCUSSION THREAD

**Discussion by LPVOIDDev** [ocornut/imgui#4021] [Old/Archived discussions]
IMHO the hardcoded `LogText(IM_NEWLINE);` ([?](https://github.com/ocornut/imgui/blob/d6a5cc7934b4f4f9d5effffc4f1acee151247f51/imgui.cpp#L10196)) inside [ImGui::LogFinish()](https://github.com/ocornut/imgui/blob/d6a5cc7934b4f4f9d5effffc4f1acee151247f51/imgui.cpp#L10196) is redundant since it's sometimes necessary to copy a text without any extra characters such as newlines
I am currently trying to create a button to copy a text to clipboard, but it shouldn't have any newlines inserted, and I found no way to do this without modifying imgui source code
I'm only 2 days to IMGUI so bear with me if I didn't notice an alternative way to accomplish this

**Comment by LPVOIDDev** [comment]
Sorry, silly me. I've just found the ImGui::SetClipboardText :)

**Comment by ocornut** [comment]
I suppose it may make sense to only add a new line in `LogFinish()` if not already on a new line?

**Reply by LPVOIDDev** [reply]
I must pretty much agree. It does make sense. Guess just checking the previous character would do the trick.
Anyways, for me as a newcomer it was unclear at first how to copy to clipboard, as I was only digging the **imgui_demo.cpp** to find an example, but I didn't dig much into **imgui.cpp**
My mistake
